packages/typespec-ts/src/utils/operationUtil.ts

```diff
@@ -437,3 +449,133 @@ export function getCollectionFormatHelper(
   const detail = getSpecialSerializeInfo(paramType, paramFormat);
   return detail.descriptions.length > 0 ? detail.descriptions[0] : undefined;
 }
+
+export async function appendDefaultResponseIfAbsent(
+  dpgContext: SdkContext,
+  client: SdkClient
+) {
+  const operationGroups = listOperationGroups(dpgContext, client);
+  const program = dpgContext.program;
+  for (const operationGroup of operationGroups) {
+    const operations = listOperationsInOperationGroup(
+      dpgContext,
+      operationGroup
+    );
+
+    for (const op of operations) {
+      const route = ignoreDiagnostics(getHttpOperationWithCache(program, op));
+      // ignore overload base operation
+      if (route.overloads && route.overloads?.length > 0) {
+        continue;
+      }
+      await appendIfAbsent(route);
+    }
+  }
+  const clientOperations = listOperationsInOperationGroup(dpgContext, client);
+  for (const clientOp of clientOperations) {
+    const route = ignoreDiagnostics(
+      getHttpOperationWithCache(program, clientOp)
+    );
+    if (route.overloads && route.overloads?.length > 0) {
+      continue;
+    }
+    await appendIfAbsent(route);
+  }
+}
+
+async function appendIfAbsent(route: HttpOperation) {
+  const errors = route.responses.filter((r) =>
+    isDefaultStatusCode(r.statusCode)
+  );
+  if (errors.length === 0) {
+    route.responses.push(FAKE_CORE_ERROR_RESPONSE as any);
+  }
+}
+
+export async function createRLCEmitterTestHost() {
+  return createTestHost({
+    libraries: [
+      HttpTestLibrary,
+      RestTestLibrary,
+      VersioningTestLibrary,
+      AzureCoreTestLibrary,
+      SdkTestLibrary
+    ]
+  });
+}
+const FAKE_CORE_ERROR_RESPONSE_TYPE = {
+  kind: "Model",
+  name: "ErrorResponse",
+  namespace: {
+    kind: "Namespace",
+    name: "Foundations",
+    namespace: {
+      kind: "Namespace",
+      name: "Core",
+      namespace: {
+        kind: "Namespace",
+        name: "Azure"
+      }
+    }
+  },
+  properties: [],
+  sourceModel: {
+    kind: "Model",
+    name: "ErrorResponseBase"
+  }
+};
+const FAKE_CORE_ERROR_RESPONSE = {
+  statusCode: "*",
+  type: FAKE_CORE_ERROR_RESPONSE_TYPE,
+  responses: [
+    {
+      body: {
+        contentTypes: ["application/json"],
```

Comment:
Is it going to be problematic if this operation doesn't have "application/json" content type at all ?

Reply:
Please notice that we will not specify the content-type as `"application/json"` in generated code because this is our fallback content type in core. This content type is more like a bypass option to let rlc not generate any content type info for this operation.

But I do re-think that generating the `ErrorModel` from core is correct or not, if the server side throws any object with their own styles that would cause issues.